feature: add docs to composite credential (#111)
UnumID/UnumID.github.io: docs/schema.mdx

We hope to soon make it possible for you as a <a href="/terminology#customer"><Tip type="customer"/></a> to define your own credential schemas. But **currently, we Verified Inc. define all credential schemas.** If you would like a new credential schema for your use case, please let us know and we will gladly define one for you.
:::

> We currently are support what can be referred to as "single attribute atomic credentials". This means that we break data into its simplest form for every credential. This is to provide users a means of selective disclosure for <a href="/terminology#request"><Tip type="request"/></a>s with optional fields.

Reviewer: ```suggestion
We currently support what can be referred to as "single attribute atomic credentials" and "composite credentials". 

Atomic credentials provide users a means of selective disclosure for <a href="/terminology#request"><Tip type="request"/></a>s with optional fields.
```

> We also support composite credentials, which are credentials that contain multiple "Atomic Credentials". For example, the `FullNameCredential` contains `FirstNameCredential`, `LastNameCredential` and `MiddleNameCredential`.

Reviewer: ```suggestion
Composite credentials are credentials that contain multiple "Atomic Credentials". For example, the `FullNameCredential` contains `FirstNameCredential`, `LastNameCredential` and `MiddleNameCredential`. This provides a means of grouping atomic credentials with adjacent data.
```

  "properties": {
    "firstName": {
      "description": "A person's first name",
      "examples": ["John", "Mary Kate"],
      "title": "First Name",
      "displayFormat": "String",
      "type": "string"
    }
  },
  "required": ["firstName"]
}
```

For composite credential schemas, the object will contain a `allOf` property, which is an array of the atomic credential schemas:

```json title="Example FullNameCredential JSON Schema"
{
  "$id": "FullNameCredential",
  "type": "object",
  "allOf": [
    {
      "$ref": "FirstNameCredential"
    },
    {
      "$ref": "LastNameCredential"
    },
    {
      "$ref": "MiddleNameCredential"
    }
  ]
}
```

:::tip
Note that each attribute has a `description` and `examples` to help you understand the credential's schema. If you need more detailed information, look at our open source [schema-sdk](https://github.com/UnumID/schema-sdk) [formats](https://github.com/UnumID/schema-sdk/blob/main/src/formats.ts).
:::

:::tip
The `$ref` property is a JSON Pointer, which is a string containing a URI fragment identifier. In this case, it is a reference to another schema. This is how we define composite credential schemas.
:::

:::note

> It's important to note that each of these credentials inside `FullNameCredential` are Atomic Credentials, so you are able to request individually.

Reviewer 2: ```suggestion
It's important to note that each of these credentials inside `FullNameCredential` are single attribute atomic credentials, so you are able to request individually.
```

> But if you need create or request a `CityCredential` and a `StateCredential` you will need to use `AddressCredential`, because you can mismatch the data.

Reviewer 2: ```suggestion
But if you need create or request a `CityCredential` and a `StateCredential` you will need to use `AddressCredential` so data doesn't get mismatched.
```

We hope to soon make it possible for you as a <a href="/terminology#customer"><Tip type="customer"/></a> to define your own credential schemas. But **currently, we Verified Inc. define all credential schemas.** If you would like a new credential schema for your use case, please let us know and we will gladly define one for you.
:::

We currently support what can be referred to as "single attribute atomic credentials" and "composite credentials". 

Atomic credentials provide users a means of selective disclosure for <a href="/terminology#request"><Tip type="request"/></a>s with optional fields.

> Composite credentials are credentials that contain multiple "Atomic Credentials". For example, the `FullNameCredential` contains `FirstNameCredential`, `LastNameCredential` and `MiddleNameCredential`. This provides a means of grouping atomic credentials with adjacent data.

Reviewer: ```suggestion
Composite credentials are credentials that contain multiple "Atomic Credentials". For example, the `FullNameCredential` contains `FirstNameCredential`, `LastNameCredential` and `MiddleNameCredential`. This provides a means of grouping atomic credentials with related data.
```

Author: hahhaa
